### PR TITLE
🏃 util/tests: standardize gomega imports

### PR DESCRIPTION
**Pull request by cpanato**
**What this PR does / why we need it**:
standardize gomega imports for `util/*/***` tests
will open other PR for the other folders to make easier the review

/cc @vincepri @detiber 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Related to https://github.com/kubernetes-sigs/cluster-api/issues/2433

### CODE REVIEW

**Comment by cpanato** [comment]
I think this was the last batch, however, when all PRs get merged i will review the tree once again

File: util/kubeconfig/kubeconfig_test.go [hunk 1]
```diff
@@ -85,19 +84,17 @@ func setupScheme() *runtime.Scheme {
...
-	if err != nil {
-		t.Fatalf("Expected no error, got %v", err)
-	}
 
-	if !reflect.DeepEqual(validSecret.Data[secret.KubeconfigDataName], found) {
-		t.Fatalf("Expected found secret to be equal to input")
-	}
+	found, err := FromSecret(context.Background(), client, clusterKey)
+	g.Expect(err).NotTo(HaveOccurred())
+	g.Expect(validSecret.Data[secret.KubeconfigDataName]).To(Equal(found))
```
Comment (JoelSpeed) [inline util/kubeconfig/kubeconfig_test.go]: Nit: Normally you `Expect` the thing you got from the function you are testing to `Equal` a known value, helps the readability when there's an error
```suggestion
	g.Expect(found).To(Equal(validSecret.Data[secret.KubeconfigDataName]))
```

File: util/kubeconfig/kubeconfig_test.go [hunk 2]
```diff
@@ -161,47 +160,25 @@ func TestNew(t *testing.T) {
...
+		g.Expect(err).NotTo(HaveOccurred())
 
 		actualConfig, actualError := New(tc.cluster, tc.endpoint, caCert, caKey)
 		if tc.expectError {
-			if actualError == nil {
-				t.Fatalf("Expected error but go nil error")
-			} else {
-				continue
-			}
+			g.Expect(actualError).NotTo(HaveOccurred())
```
Comment (JoelSpeed) [inline util/kubeconfig/kubeconfig_test.go]: I think this should be an expectation for the error to have occurred looking at the original semantics 🤔 WDYT?
```suggestion
			g.Expect(actualError).To(HaveOccurred())
```

File: util/kubeconfig/kubeconfig_test.go [hunk 3]
```diff
@@ -161,47 +160,25 @@ func TestNew(t *testing.T) {
...
-				tc.cluster, tc.expectedConfig.Contexts[tc.expectedConfig.CurrentContext].AuthInfo,
-				actualConfig.Contexts[tc.expectedConfig.CurrentContext].AuthInfo)
-		}
+		g.Expect(actualConfig.Clusters).To(HaveLen(len(tc.expectedConfig.Clusters)))
+		g.Expect(actualConfig.Contexts).To(HaveLen(len(tc.expectedConfig.Contexts)))
+
+		g.Expect(actualConfig.Clusters[tc.cluster]).NotTo(BeNil())
+		g.Expect(actualConfig.Contexts[tc.expectedConfig.CurrentContext]).NotTo(BeNil())
+		g.Expect(actualConfig.CurrentContext).To(Equal(tc.expectedConfig.CurrentContext))
+		g.Expect(actualConfig.Contexts).To(Equal(tc.expectedConfig.Contexts))
```
Comment (JoelSpeed) [inline util/kubeconfig/kubeconfig_test.go]: Could the order vary of `actualConfig.Contexts`? Might be better to use `ConsistOf` if ordering doesn't matter
```suggestion
		g.Expect(actualConfig.Contexts).To(ConsistOf(tc.expectedConfig.Contexts))
```

File: util/kubeconfig/kubeconfig_test.go (hunk 2 above)
Comment (cpanato) [reply util/kubeconfig/kubeconfig_test.go]: right!

File: util/kubeconfig/kubeconfig_test.go (hunk 3 above)
Comment (cpanato) [reply util/kubeconfig/kubeconfig_test.go]: if use `ConsistOf` the test fail
```
--- FAIL: TestNew (0.29s)
    /Users/cpanato/go/src/github.com/kubernetes-sigs/cluster-api/util/kubeconfig/kubeconfig_test.go:180: 
        Expected
            <map[string]*api.Context | len:1>: {
                "foo-admin@foo": {
                    LocationOfOrigin: "",
                    Cluster: "foo",
                    AuthInfo: "foo-admin",
                    Namespace: "",
                    Extensions: nil,
                },
            }
        to consist of
            <[]interface {} | len:1, cap:1>: [
                {
                    "foo-admin@foo": {
                        LocationOfOrigin: "",
                        Cluster: "foo",
                        AuthInfo: "foo-admin",
                        Namespace: "",
                        Extensions: nil,
                    },
                },
            ]
        the missing elements were
            <[]interface {} | len:1, cap:1>: [
                {
                    "foo-admin@foo": {
                        LocationOfOrigin: "",
                        Cluster: "foo",
                        AuthInfo: "foo-admin",
                        Namespace: "",
                        Extensions: nil,
                    },
                },
            ]
        the extra elements were
            <[]interface {} | len:1, cap:1>: [
                {
                    LocationOfOrigin: "",
                    Cluster: "foo",
                    AuthInfo: "foo-admin",
                    Namespace: "",
                    Extensions: nil,
                },
            ]
FAIL
FAIL	sigs.k8s.io/cluster-api/util/kubeconfig	0.715s
FAIL
Error: Tests failed.
```

Comment (JoelSpeed) [reply util/kubeconfig/kubeconfig_test.go]: Ack. Ignore that suggestion 😅 Don't think it works for maps

**Pull request merged**